### PR TITLE
Color by gene set mean expression

### LINKED ISSUE #2036: As an eBio or cBio, I can visualize the mean expression of a gene set
Calculate the mean gene expression for the gene set, render the mean expression for the set, enable coloring by.

### PR DESCRIPTION
**Pull request by colinmegill**
Fixes #2036

### CODE REVIEW

**Comment by signechambers1** [comment]
Undo / redo does not work for color by for mean expression of a gene set. Colin is fixing.

File: client/src/components/geneExpression/menus/genesetMenus.js [hunk 1]
```diff
@@ -43,19 +46,41 @@ class GenesetMenus extends React.PureComponent {
...
+
+    const isColorBy = geneset === colorAccessor;
+    const genesetClosed = !isOpen;
+    const showingAllGenes = !toggleSummaryHisto;
+
+    const notShowingSummary = genesetClosed || showingAllGenes;
 
     return (
       <>
         {genesetsEditable ? (
```
Comment (bkmartinjr) [inline client/src/components/geneExpression/menus/genesetMenus.js:81]: suggest using the preferred style:  `genesetsEditable && (...)` (ie, don't use ternary if the third term is a null)

File: client/src/util/stateManager/colorHelpers.js [hunk 2]
```diff
@@ -33,6 +34,27 @@ export function createColorQuery(colorMode, colorByAccessor, schema) {
...
       ];
     }
+    case "color by geneset mean expression": {
+      const varIndex = schema?.annotations?.var?.index;
+
+      if (!varIndex) return null;
+      if (!genesets) return null;
+
+      const _geneset = genesets.get(colorByAccessor);
+      const _setGenes = Array.from(_geneset.genes.keys());
```
Comment (bkmartinjr) [inline client/src/util/stateManager/colorHelpers.js:44]: Suggest that this would be clearer:    
> const _setGenes = [ ..._geneset.genes ];

**Review by bkmartinjr — APPROVED** [review]
I did a code-only review (not UX).  LGTM

**Comment by maniarathi** [comment]
@seve Can you do the UX review please?

**Comment by maniarathi** [comment]
Oops -- I misinterpreted Bruce's comment. Removing Seve from the review list :)

**Review by signechambers1 — APPROVED** [review]

**Pull request merged**